paddle/fluid/operators/activation_op.cc

```diff
@@ -83,6 +83,7 @@ class ActivationGradOpDescMaker : public framework::SingleGradOpDescMaker {
     op->SetInput(framework::GradVarName("Out"), OutputGrad("Out"));
     op->SetOutput(framework::GradVarName("X"), InputGrad("X"));
     op->SetAttrMap(Attrs());
+    op->SetAttr("forward_input_X", Input("X").back());
```

Comment:
Do you mean
```
if (op->HasAttr("use_mkldnn"))
   op->SetAttr("forward_input_X", Input("X").back());
```
by https://github.com/PaddlePaddle/Paddle/issues/18788#issuecomment-514599133 @jczaja

Reply:
More like:
```cpp
if (executing by mkldnn) {
    op->SetInput("X", Input("X"));
}
````

Reply:
why `SetInput("X", Input("X"))` here? not `op->SetAttr("forward_input_X", Input("X").back());`?

Reply:
So essentially
```C++
if (op->HasAttr("use_mkldnn") && op->Attr<bool>(use_mkldnn))
   op->SetAttr("forward_input_X", Input("X").back());
```

Reply:
@sneaxiy How do you think about https://github.com/PaddlePaddle/Paddle/pull/18762#discussion_r306772070. I think it doesn't affect Memory optimization strategy in both training and inference.

Reply:
I just realised that since `use_mkldnn` is set during MKLDNN Placement Pass, the `use_mkldnn` attribute will return false here. We need to find another way of determining if MKLDNN is to be enabled here.

Reply:
@luotao1 `SetInput()` instead of `SetAttr()` because you didn't like additional attributes and it carries exact same information. As long as we use mkldnn not gpu we won't mess with gpu memory strategy and it's also easier for us to reference it as Input.

Reply:
As this is training via python, how about using FLAGS_use_mkldnn value to determine if we should add another input or not ?